### PR TITLE
[EFM] KV Store supports mutating `EpochExtensionViewCount`

### PR DESCRIPTION
**Pull request by durkmurder**
https://github.com/onflow/flow-go/issues/6227

### Context 

This PR implements part of the referenced issue, specifically: 

> - Implement `SetEpochExtensionViewCount` and a sanity check for `FinalizationSafetyThreshold`
> - Add tests to ensure we meet the sanity check. Suggested in https://github.com/onflow/flow-go/pull/6209#discussion_r1680392404.

Previously we were relying on `defaultEpochExtensionViewCount` but to allow flexible configuration for different chain-ids I have moved the default value to the sporking code and enforced KV store configuration using constructor.

### CODE REVIEW

**Comment by durkmurder** [comment]
@zhangchiqing FYI, I have added an extra flag to the sporking process

File: state/protocol/defaults.go [hunk 1]
```diff
@@ -6,16 +6,27 @@ import (
...
-		return 1_000, nil
+		return SafetyParams{
+			FinalizationSafetyThreshold: 1_000,
+			EpochExtensionViewCount:     100_000, // approximately 1 day
+		}, nil
 	case flow.Localnet, flow.Benchnet, flow.BftTestnet, flow.Emulator:
-		return 100, nil
+		return SafetyParams{
+			FinalizationSafetyThreshold: 100,
+			EpochExtensionViewCount:     690, // approximately 10 minutes
```
Comment (jordanschalm) [inline state/protocol/defaults.go:28]:
```suggestion
			EpochExtensionViewCount:     600, // approximately 10 minutes
```

File: cmd/bootstrap/cmd/rootblock.go [hunk 2]
```diff
@@ -93,12 +94,14 @@ func addRootBlockCmdFlags() {
...
 	rootBlockCmd.Flags().UintVar(&flagProtocolVersion, "protocol-version", flow.DefaultProtocolVersion, "major software version used for the duration of this spork")
 	rootBlockCmd.Flags().Uint64Var(&flagEpochCommitSafetyThreshold, "epoch-commit-safety-threshold", 500, "defines epoch commitment deadline")
+	rootBlockCmd.Flags().Uint64Var(&flagEpochExtensionViewCount, "epoch-extension-view-count", 100_000, "length of epoch extension in views, default is 100_000 which is approximately 1 day")
 
 	cmd.MarkFlagRequired(rootBlockCmd, "root-chain")
 	cmd.MarkFlagRequired(rootBlockCmd, "root-parent")
 	cmd.MarkFlagRequired(rootBlockCmd, "root-height")
 	cmd.MarkFlagRequired(rootBlockCmd, "protocol-version")
 	cmd.MarkFlagRequired(rootBlockCmd, "epoch-commit-safety-threshold")
+	cmd.MarkFlagRequired(rootBlockCmd, "epoch-extension-view-count")
```
Comment (jordanschalm) [inline cmd/bootstrap/cmd/rootblock.go:104]: Noting for both these that, for subsequent sporks, we won't want them to be required inputs (they should instead be pulled in from the prior spork's KV Store). We can figure out the details in https://github.com/onflow/flow-go/issues/5973 though.

**Review by jordanschalm — APPROVED** [review]

**Comment by jordanschalm** [comment]
Once we've finalized this PR, and that it will go out for Mainnet25, let's add the new flag to https://github.com/dapperlabs/dapper-flow-hosting/pull/1449.

**Review by kc1116 — APPROVED** [review]

**Pull request merged**